include/deal.II/lac/petsc_matrix_base.h

```diff
@@ -826,6 +863,13 @@ namespace PETScWrappers
      */
     operator Mat () const;
 
+    /**
+     * Return a reference to the underlying PETSc type. It can be used to
+     * modify the underlying data, so use it only when you know what you
+     * are doing.
+     */
+    Mat &petsc_matrix ();
```

Comment:
out of curiosity, why do you need this? Why is using the conversion operator above not sufficient?

Reply:
I thought the conversion operator could only be used to access the `Mat`, not to modify it. Besides, how do I use the conversion operator to get a pointer to the underlying `Mat`?

Reply:
you can use the conversion operator by doing things like `static_cast<const Mat &>(petsc_wrapper_object)`. 

Where are you using the non-`const` version of the operator? I'm not overly fond of having this operator because it makes it easy to introduce subtle bugs where someone gets the `Mat` object, makes changes to it so that it now represents something different, but the other member variables of the PETSc wrapper object are not updated in the same way. These are subtle bugs that are difficult to find later on, and we try hard to stick with class designs that make it difficult to introduce these sorts of bugs.

I don't want to suggest that you don't need this operator, but I'd really like to understand where it is necessary and whether we can work around it.

Reply:
What was the conclusion here?